reark/src/main/java/io/reark/reark/data/DataStreamNotification.java

```diff
@@ -67,28 +76,42 @@ public T getValue() {
     }
 
     @Nullable
-    public Throwable getError() {
-        return error;
+    public String getErrorBody() {
+        return errorBody;
+    }
+
+    public int getHttpCode() {
+        return httpCode;
+    }
+
+    @Nullable
+    public String getErrorMessage() {
+        return errorMessage;
     }
 
     @NonNull
-    public static<T> DataStreamNotification<T> fetchingStart() {
+    public static <T> DataStreamNotification<T> fetchingStart() {
         return new DataStreamNotification<>(Type.FETCHING_START, null, null);
     }
 
     @NonNull
-    public static<T> DataStreamNotification<T> onNext(T value) {
-        return new DataStreamNotification<>(Type.ON_NEXT, value, null);
+    public static <T> DataStreamNotification<T> onNext(T value) {
+        DataStreamNotification<T> data = new DataStreamNotification<>(Type.ON_NEXT, value, null);
+        data.httpCode = 200;
```

Comment:
[Here](https://gist.github.com/tomaszpolanski/9db274967247dd87028cee68d6d61534) is an example how I would refactor this class. 
The biggest benefit is that the class is now immutable.